### PR TITLE
[formrecognizer] rename display_name -> model_name

### LINKED ISSUE #14373: [formrecognizer] rename display_name to model_name

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves #14373 

Updated APIView: https://apiview.dev/Assemblies/Review/36ac7c8c633d4bf6a45ee8672243f195?diffRevisionId=26737bcc186542c69b0ac35155d27181&doc=False

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 1]
```diff
@@ -286,13 +286,13 @@ form_training_client = FormTrainingClient(endpoint, credential)
...
 poller = form_training_client.begin_training(
-    container_sas_url, use_training_labels=False, display_name="my first model"
+    container_sas_url, use_training_labels=False, model_name="my first model"
 )
 model = poller.result()
 
 # Custom model information
 print("Model ID: {}".format(model.model_id))
-print("Display name: {}".format(model.display_name))
+print("Display name: {}".format(model.model_name))
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:295]:
```suggestion
print("Model name: {}".format(model.model_name))
```
maybe?

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:295]: in other places too

Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:295]: 😱  I thought I got them all, thanks!

**Review by maririos — APPROVED** [review]

**Review by iscai-msft — APPROVED** [review]
:shipit:

**Review by maririos — APPROVED** [review]

**Pull request merged**